internal/redact/redact_test.go

```diff
@@ -0,0 +1,251 @@
+package redact
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+func ExampleError() {
+	// Each error string in a chain of wrapped errors is redacted with a
+	// placeholder describing the error's type.
+	wrapped := errors.New("not found")
+	name := "Alex"
+	err := fmt.Errorf("error getting user %s: %w", name, wrapped)
+
+	fmt.Println("Normal:", err)
+	fmt.Println("Redacted:", Error(err))
+	// Output:
+	// Normal: error getting user Alex: not found
+	// Redacted: <redacted *fmt.wrapError>: <redacted *errors.errorString>
+}
+
+func ExampleErrorf() {
+	// Errors created with Errorf are redacted by omitting any arguments not
+	// marked as safe. The literal portion of the format string is kept.
+	wrapped := errors.New("not found")
+	name := "Alex"
+	id := 5
+	err := Errorf("error getting user %s with ID %d: %w", name, Safe(id), wrapped)
+
+	fmt.Println("Normal:", err)
+	fmt.Println("Redacted:", Error(err))
+	// Output:
+	// Normal: error getting user Alex with ID 5: not found
+	// Redacted: error getting user <redacted string> with ID 5: <redacted *errors.errorString>
+}
+
+func ExampleError_wrapped() {
+	// If an error wraps another, then redacting it results in a message with the
+	// redacted version of each error in the chain up until the first redactable
+	// error.
+	name := "Alex"
+	err := fmt.Errorf("fatal error: %w",
+		Errorf("error getting user %s: %w", name,
+			errors.New("not found")))
+
+	fmt.Println("Normal:", err)
+	fmt.Println("Redacted:", Error(err))
+	// Output:
+	// Normal: fatal error: error getting user Alex: not found
+	// Redacted: <redacted *fmt.wrapError>: error getting user <redacted string>: <redacted *errors.errorString>
+}
+
+func TestNil(t *testing.T) {
+	checkUnredactedError(t, nil, "<nil>")
+	checkRedactedError(t, nil, "<nil>")
+}
+
+func TestSimple(t *testing.T) {
+	err := errors.New("simple")
+	checkUnredactedError(t, err, "simple")
+	checkRedactedError(t, err, "<redacted *errors.errorString>")
+}
+
+func TestSimpleWrapSimple(t *testing.T) {
+	wrapped := errors.New("error 2")
+	err := fmt.Errorf("error 1: %w", wrapped)
```

Comment:
does this also work with `errors.Wrap(err, "some wrapping msg")`  from `github.com/pkg/errors`?